### ISSUE TITLE
Payment Page: Pay with Xumm: Dollar Amount and XRP Amount Don't Agree

### ISSUE THREAD

**Issue by NetScr1be** [KevinKoobs/xBuyTangem-issues#8]
### Issue Description

In the Pay with Xumm Button it says 10 XRP;

- no matter what Reserve Coverage Quantity is selected
- at current prices it should be  $XRP 115.xx

![image](https://github.com/KevinKoobs/xBuyTangem-issues/assets/5115007/16114ebd-743f-41dd-ade3-a891428de815)

**ALSO: Despite the text above describing CC payments as more expensive CC Payment amount is less** 

Amount updates once Pay with Xumm button is selected but the transaction is denominated in Euros (default currency for this instance is CAD). If we're paying with Xumm why isn't it XRP?

![image](https://github.com/KevinKoobs/xBuyTangem-issues/assets/5115007/dce8a843-dd4c-4dea-8cbe-2781d165e2ff)

_Slide to Accept_ is greyed out

The only option is to _Find new payment options_

Selecting _Find new payment options_ causes xApp to switch screens for a few seconds then comes back to _No payment options found_.

The account has adequate XRP balance for payment.

### Steps to Reproduce

1. Open xApp
2. Select Order Cards
3. Select Quantity
4. Select Reserve Coverage
5. Select Shipping Country
6. Continue to payment

### Error Messages

N/A

### Account Address

rpGhdshAMiVYPei3FtusVTPndijbpdATna

### Device Info

- Pixel 6 Pro
- Android 14 (Build UP1A.231005.007)

**Comment by WietseWind** [comment]
@KevinKoobs 

<img width="650" alt="image" src="https://github.com/KevinKoobs/xBuyTangem-issues/assets/4756161/db77c5c2-6ac7-4751-b797-9038cc86ee7d">

For one thing, this is not OK, the On Ledger payments can and should never be more expensive than a card payment.

**Comment by WietseWind** [comment]
@NetScr1be 

Two questions:
1. Certain you had `rpGhdshAMiVYPei3FtusVTPndijbpdATna` active when doing this?
2. When you try again now, does it perhaps work?

Sometimes pathfinding falls over (sucks, technical debt in XRPL)

If it keeps on not working, we'll have to look at the payload. If it works now, it was a temp error.

https://xrpl-pathfinding.netlify.app/ (debug tool) now finds paths.
https://htzdk2l.dlvr.cloud/pasted_2.png
https://htzdk2l.dlvr.cloud/pasted_1.png

**Comment by NetScr1be** [comment]
> 1.  Certain you had `rpGhdshAMiVYPei3FtusVTPndijbpdATna` active when doing this?  
>       

 Just reproduced it using that account.
 
 Same failure.

**Comment by NetScr1be** [comment]
> the transaction is denominated in Euros (default currency for this instance is CAD). **If we're paying with Xumm why isn't it XRP?**

It's baffling how Euros enter into this at all.

**Comment by WietseWind** [comment]
> > the transaction is denominated in Euros (default currency for this instance is CAD). **If we're paying with Xumm why isn't it XRP?**
> 
> It's baffling how Euros enter into this at all.

Not at all, we create a payload to get paid in EUR, so you'll end up delivering EUR, that's by design. The pathfinding is for the source currency, the destination currency is predefined, EUR, as we want it. Works that way for ages with for example the Vanity purchases.

**Comment by WietseWind** [comment]
> > the transaction is denominated in Euros (default currency for this instance is CAD). **If we're paying with Xumm why isn't it XRP?**
> 
> It's baffling how Euros enter into this at all.

Are you doing this on a 2.6 Beta install? Do you have a phone around without the Beta with that account on it? I think it may be a 2.6 beta bug.

**Comment by KevinKoobs** [comment]
> @KevinKoobs
> 
> <img alt="image" width="650" src="https://user-images.githubusercontent.com/4756161/283419794-db77c5c2-6ac7-4751-b797-9038cc86ee7d.png">
> For one thing, this is not OK, the On Ledger payments can and should never be more expensive than a card payment.

Fixed in https://github.com/XRPL-Labs/xBuyTangem/commit/8bfc9f4d8a6f3cf11e010041ade489882e46238a
